python/triton/backends/compiler.py

```diff
@@ -135,18 +149,18 @@ def hash(self):
         return hashlib.sha256(key.encode("utf-8")).hexdigest()
 
     def to_dict(self):
-        return self.arg_properties
+        return {"arg_properties": self.arg_properties, "cls": type(self).__name__}
```

Comment:
I'd suggest to put a comment here to explain that "We are also recording the (sub)class name here so we can reconstruct the same (sub)class with the reverse from_dict()."

Or actually maybe you can put this comment in the `descriptor_table` comment.